go/lang/security/injection/tainted-url-host.yaml

```
@@ -53,4 +40,53 @@ rules:
     - metavariable-regex:
         metavariable: $ANYTHING
         regex: ^(BasicAuth|Body|Cookie|Cookies|Form|FormValue|GetBody|Host|MultipartReader|ParseForm|ParseMultipartForm|PostForm|PostFormValue|Referer|RequestURI|Trailer|TransferEncoding|UserAgent|URL)$
+  - label: FORMAT
+    requires: INPUT
+    patterns:
+    - pattern-either:
+      - patterns:
+        - pattern-either:
+          - pattern: fmt.Fprintf($F, "$URLSTR", $INPUT, ...)
+          - pattern: fmt.Sprintf("$URLSTR", $INPUT, ...)
+          - pattern: fmt.Printf("$URLSTR", $INPUT, ...)
+        - metavariable-regex:
+            metavariable: $URLSTR
+            regex: ^((http://)|(https://)|(//))%(v|s|q).*
+    - focus-metavariable: $INPUT
+  - label: ADD_OR_DIRECT
+    requires: INPUT
+    patterns:
+    - pattern-either:
+      - pattern: $X + $F
+      - pattern: $X
+    - focus-metavariable: $X
+  pattern-sanitizers:
+  - patterns:
+    - pattern: $F + $X
+    - metavariable-regex: 
+        metavariable: $F
+        regex: ".*//[a-zA-Z0-10]+..*"
```

```suggestion
        regex: ".*//[a-zA-Z0-10]+\\..*"
```
I'm assuming the `.` after the greedy quantifier is meant to be an explicit `.` which means you're missing an escape here

---

ah yes it definitely is meant to be explicit `.` thank you!

---

added some not-insides to exclude cases where the direct `$X` matches  a fmt-string line and proceeds to override the cases where `fmt.Println("example.com%q)` are not matched